repositories/migrations/20240402153800_audit_trail_trigger.sql

```diff
@@ -0,0 +1,55 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TYPE audit_operation AS ENUM('INSERT', 'UPDATE', 'DELETE');
+
+-- CreateTable
+CREATE TABLE
+	audit (
+		"id" UUID NOT NULL DEFAULT gen_random_uuid () PRIMARY KEY,
+		"operation" audit_operation NOT NULL,
+		"user_id" TEXT,
+		"table" VARCHAR NOT NULL,
+		"entity_id" UUID NOT NULL,
+		"data" JSONB NOT NULL DEFAULT '{}',
+		"created_at" TIMESTAMPTZ (6) NOT NULL DEFAULT CURRENT_TIMESTAMP
+	);
+
+-- Order audit trigger function
+CREATE
+OR REPLACE FUNCTION global_audit () RETURNS TRIGGER AS $$
+    BEGIN
+        IF (TG_OP = 'DELETE') THEN
```

Comment:
`TG_*` contains some metadata on the trigger event, similarly `NEW` and `OLD` hold the data of the row acted on (with obvious naming).